### DISCUSSION TITLE
Customized Pages

### DISCUSSION THREAD

**Discussion by TwixtedChaox** [codeigniter4/shield#532] [Show and tell]
Customized the Login, Register, and Magic-Link pages

Login Screen
![image](https://user-images.githubusercontent.com/3359780/202687813-43e361a7-f1b5-4540-8f72-ff9ec3e8fe2c.png)

Register Screen
![image](https://user-images.githubusercontent.com/3359780/202688003-e2d5ed17-d47e-4299-a815-efc9ce356df6.png)

Magic Link Screen
![image](https://user-images.githubusercontent.com/3359780/202688082-637dfede-bf7e-480d-8772-7ce3fce210de.png)

**Comment by MGatner** [comment]
Beautiful! Did you do this in a framework?

**Reply by TwixtedChaox** [reply]
I used a bit of Bootstrap, but mostly just simple css design.
I did utilize Glyphicons as well.

**Comment by Salman937** [comment]
@TwixtedChaox  how did you Customize it share some tips. Thank you

**Reply by lavahasif** [reply]
![image](https://user-images.githubusercontent.com/22430922/203976406-07d7c4cd-8d4c-45f9-9ecd-48b539454958.png)

![image](https://user-images.githubusercontent.com/22430922/203976714-e7cd3b44-f17e-4b9b-b46d-1b1019bab56a.png)

~You can goto (vendor folder )package folder and easily modify views~

**Reply by grisinformatica** [reply]
> @TwixtedChaox how did you Customize it share some tips. Thank you

Hi, just create your own views and edit [Auth config file](https://github.com/codeigniter4/shield/blob/develop/src/Config/Auth.php).

**Reply by gphg** [reply]
I've been working with this library. And what @lavahasif instructed by modify the view file under the vendor is not advised. It works, but any changes made under the vendor aren't tracked by git and likely to be updated by composer.

I suggest you to copy the proved views onto your application views directory (`app/Views`), then edit the `config/Auth.php`, look up for `$views` and modify with your own.

This issues: https://github.com/codeigniter4/shield/issues/545 might give a hint. It contains several instructions how to extend the library without touched the original file.

**Reply by TwixtedChaox** [reply]
Please do NOT do as lavahasif instructed. Yes it is a great place to get the code needed for your view file but I created my own views for my app and instructed to app to use those views as shown by grisinformatica.

Editing the default views WILL be overwritten if you do any updates at all to your application and dependencies.
It is always best practice to create your own (even if by simply copying the views files into your apps views folder) and instructing the appropriate views to be used.

As gphg stated, (I wish i could upvote his answer) THIS is the correct way to work with this library.
